### PR TITLE
CActiveAEResamplePi: remove usage of videoplayer.synctype

### PR DESCRIPTION
**Pull request by Montellese**
This removes the usage of the `videoplayer.synctype` after df1c1b8d25f8ae227ee76399f6ef7b6b98f34201.

I can't build/test this and I don't know this code. @FernetMenta: Let me know if this is ok.

### CODE REVIEW

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEResamplePi.cpp [hunk 1]
```diff
@@ -280,7 +280,7 @@ bool CActiveAEResamplePi::Init(uint64_t dst_chan_layout, int dst_channels, int d
 
   LOGTIME(3);
 
-  if (CSettings::Get().GetBool("videoplayer.usedisplayasclock") && CSettings::Get().GetInt("videoplayer.synctype") == SYNC_RESAMPLE)
+  if (CSettings::Get().GetBool("videoplayer.usedisplayasclock"))
```
Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEResamplePi.cpp]: I don't know why this is needed at all. @popcornmix AE does not instantiate a resampler for any case a resampler is not needed. why did you add this check?

Comment (popcornmix) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEResamplePi.cpp]: Pretty sure we do get called for format/interleaved/planar conversion without actual resampling.

Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEResamplePi.cpp]: that may be true but the dependency to the settings makes no sense. why don't you check m_src_rate and m_dst_rate

Comment (popcornmix) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEResamplePi.cpp]: The GPU resampler stores context when resampling.
It needs to know the difference between never resampling (i.e. bypass the resample code), and resampling currently by a factor of 1.0, but may change in the future.

m_src_rate and m_dst_rate may start off as equal, but I need to know if Resample() will ever be called with a ratio != 1.0 in the future.

Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEResamplePi.cpp]: you take assumptions which may not hold in the future or even now. this broke because of a change in a part of the code which should not have a relationship to this. paplayer will make use of resampling too. 
check AESTREAM_FORCE_RESAMPLE. AE could pass this flag to resampler if required.

**Comment by Montellese** [comment]
I just saw that you have already fixed this in #6641.
